src/api/mod.rs

```diff
@@ -77,14 +77,15 @@ pub struct AsyncElementLink< E: AsyncElement> {
 
 impl<E: AsyncElement> AsyncElementLink<E> {
     pub fn new(input_stream: ElementStream<E::Input>, element: E, queue_capacity: usize) -> Self {
+        assert!( queue_capacity <= 1000, format!("Async Element queue_capacity: {} > 1000", queue_capacity));
```

Comment:
I think this commit is useful, but should it be in this PR?

Reply:
Hmm, that was probably added slightly by mistake.  Will fork that out somewhere else if at all possible.

Reply:
Removed this to be added back in on a later PR